weave/trace_server/calls_query_builder/cte.py

```diff
@@ -0,0 +1,42 @@
+"""Common Table Expression (CTE) definitions for query building."""
+
+from pydantic import BaseModel, Field
+
+
+class CTE(BaseModel):
+    """Represents a single Common Table Expression (CTE)."""
+
+    name: str
+    sql: str
+
+    def to_sql(self) -> str:
+        """Format this CTE for inclusion in a WITH clause."""
+        return f"{self.name} AS ({self.sql})"
+
+
+class CTECollection(BaseModel):
```

Comment:
Thoughts on inheriting this from `list[CTE]` directly?

Reply:
Yeah let me try that